tests/upgrades/test_syncplan.py

```diff
@@ -0,0 +1,110 @@
+
+:CaseLevel: Acceptance
+
+:CaseComponent: API
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+
+
+from nailgun import entities
+from robottelo.test import APITestCase
+from upgrade_tests import post_upgrade, pre_upgrade
+from fauxfactory import gen_string
+from upgrade_tests.helpers.scenarios import (
+    create_dict,
+    get_entity_data)
+from robottelo.decorators import skip_if_bug_open
+
+
+class ScenarioSyncPlan(APITestCase):
+    """The test class contains pre-upgrade and post-upgrade scenarios to test
+    sync_plan migration from pulp to katello
+
+    Test Steps:
+
+    1. Before Satellite upgrade, Perform test for
+       sync plan create, get
+    2. Upgrade satellite.
+    3. Perform tests for to verify the sync plans, after upgrade.
+    4. Check if tests passed.
+
+    """
+    @classmethod
+    def setUpClass(cls):
+        """Create an organization which can be re-used in tests."""
+        super(ScenarioSyncPlan, cls).setUpClass()
+        cls.sync_plan_name = "Test_Sync_plan_Migration"+gen_string('alpha')
+
+    @skip_if_bug_open('bugzilla', 1646988)
+    @pre_upgrade
+    def test_pre_sync_plan_migration(self):
+        """Pre-upgrade scenario that tests sync plan create, get, assign before
+        satellite can upgrade with pulp.
+
+        :id: badaeec2-d42f-41d5-bd85-4b23d6d5a724
+
+        :steps:
+            1. Create Repository
+            2. Create Sync Plan
+            3. Assign Sync Plan and Sync the repo
+            4. verify it passe all checks
+
+        :expectedresults: Run sync plan create, get, assign and verify it passes
+
+         """
+        self.org = entities.Organization().create()
+        sync_plan = entities.SyncPlan(
+            organization=self.org,
+            name=self.sync_plan_name
+            ).create()
+        product = entities.Product(organization=self.org).create()
+        entities.Repository(product=product).create()
+        sync_plan.add_products(data={'product_ids': [product.id]})
+        product.sync()
+        product = product.read()
+        self.assertEqual(product.sync_plan.id, sync_plan.id)
+        sync_plan = sync_plan.read()
+        self.assertEqual(sync_plan.name, self.sync_plan_name)
+        global_dict = {self.__class__.__name__: {
+            'sync_plan_name': self.sync_plan_name,
+            'interval': sync_plan.interval,
+            'sync_date': sync_plan.sync_date,
+            'product_id': product.id,
+            'sync_plan_id': sync_plan.id,
+            'org_id': self.org.id
+        }}
+        create_dict(global_dict)
+
+    @skip_if_bug_open('bugzilla', 1646988)
+    @post_upgrade
+    def test_post_sync_plan_migration(self):
+        """Post-upgrade scenario that tests existing sync plans are working as
+        expected after satellite upgrade with migrating to katello
+
+        :id: badaeec2-d42f-41d5-bd85-4b23d6d5a724
+
+        :steps:
+            1. verify it passe all checks with existing sync plans
```

Comment:
> @omkarkhatavkar looks good... can we extend post upgrade scenario to update and delete the same sync plan after upgrade? It would be great.

make sense